RFCs/0028-BlameCollector-Hang-Detection.md

```diff
@@ -1,44 +1,106 @@
-                <Configuration>
-                    <CollectDumpOnTestSessionHang TestTimeout="300000" DumpType="mini" />
-                    <ResultsDirectory>%AGENT_TEMPDIRECTORY%\TestResults</ResultsDirectory>
-                </Configuration>
-            </DataCollector>
-        </DataCollectors>
-    </DataCollectionRunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="blame" enabled="True">
+        <Configuration>
+          <CollectDumpOnTestSessionHang TestTimeout="300000" DumpType="mini" />
+          <ResultsDirectory>%AGENT_TEMPDIRECTORY%</ResultsDirectory>
+        </Configuration>
+      </DataCollector>
+    </DataCollectors>
+  </DataCollectionRunSettings>
 </RunSettings>
 ```
 
-Note: This works along with (but can also be used independent of) the existing collect dump option introduced in https://github.com/microsoft/vstest-docs/blob/master/RFCs/0024-Blame-Collector-Options.md
+Note: This works along with (but can also be used independent of) the existing collect dump option introduced in [RFC 0024](0024-Blame-Collector-Options.md).
 
 ## Supported options
-DumpType: If you choose to collect a full process dump. It takes values mini/full. By default, a mini dump will be created.
-TestTimeout: Duration of inactivity in milliseconds (no test events from the test host) after which the data collector assumes a hang has occurred and proceeds to collect a dump and kill the test host process.
-ResultsDirectory: Location to temporarily store the collected process dump
 
-## VsTest task usage
+|Option|Description|
+|--|--|
+`DumpType`|If you choose to collect a full process dump. It takes values `mini`/`full`. By default, a mini dump will be created.
+`TestTimeout`|Duration of inactivity in milliseconds (no test events from the test host) after which the data collector assumes a hang has occurred and proceeds to collect a dump and kill the test host process.
+`ResultsDirectory`|The path must exist, but files will not be permanently stored there. Test results ultimately go to a `TestResults` folder under the test project directory by default, or where the `--results-directory` parameter specifies.
+
+## Invoking the tests
+
+Add the `--settings path-to\test.runsettings` parameter to `dotnet test` or `vstest.console.exe`.
+Dump collection with Blame data collector is only supported on Windows for now. For non-Windows agents, omit the .runsettings file or use one that does not activate the Blame collector.
+
+Use the `--results-directory` to control where the `Sequence_<guid>.xml` and `.dmp` files are placed.
+
+## Outputs
+
+The `Sequence_guid.xml` and .dmp files will be placed under the test project source directory in a `TestResults\<guid>\` folder.
+
+Note that the test runner may report a failure to produce the dump file, such as the one shown below:
+
+```
+Data collector 'Blame' message: System.IO.FileNotFoundException: Collect dump was enabled but no dump file was generated.
```

Comment:
Also this is probably coming from the OG blame and not the hang detection part. I took a look at the runsettings you are using and it has the `<CollectDump DumpType="full" />` node which also configures crash dumps.

Reply:
Would you like me to remove all traces of crash dump tips? I could potentially move them to the other RFC. 
IMO though, anyone who goes to the trouble to configure dump collection at all *probably* wants both hang and crash dumps. So combining the two into one doc with a few notes to say "you can turn this individual feature on/off like this..." makes the most sense.

Reply:
I would prefer if we did this. Maybe in a separate pr.

The reason being crash detection mode has a perf hit associated with it that scales linearly with the number of exceptions being thrown by the test code. Proc dump has a first chance hook on these and even if they are handled exceptions there's a perf hit.

Reply:
ProcDump has managed hooks? I had no idea. A quick review of the options confirms what you say, but I'm disappointed if the perf hit of first chance exception monitoring applies even when you *don't* pass in the switches to enable that behavior.

Reply:
Sadly this is true, in fact when this was initially written folks didn't realize this and the crash detection was enabled by default. Once it got enabled for all customers we got reports that there is a sudden perf hit. 

I investigated the cause and realized this was it. And that the switch that says "include first chance exceptions" just indicates whether to write the dump to disk when it occurs not whether it should monitor first chance exceptions or not. But it made sense because that's the only way proc dump would be able to capture the state of the process as is.

If it had relied on unhandledException hook then the stack would have unwound by the time the proc dump got notified.